File: GameStudies/CPP/NPC.cpp
#include "NPC.hpp"
#include <iostream>

NPC::NPC(std::string nome) : Personagem(nome) {}

void NPC::adicionarDialogo(std::string fala) {
    dialogos.push_back(fala);
}

void NPC::setQuest(Quest* q) {
    quest = q;
}

void NPC::interagir() {
    if (quest->getStatus() == INATIVA) {
        std::cout << "Aceite esta missão!\n";
        quest->iniciar();
    }
    else if (quest->getStatus() == COMPLETA) {
        std::cout << "Missão completa! Recompensa entregue.\n";
        quest->finalizar();
    }
}

File: GameStudies/CS/NPC.cs
using UnityEngine;

public class NPC : Personagem
{
    public string funcao; // vendedor, guia, etc.
    public bool hostil = false;

    public void Interagir()
    {
        if (hostil)
        {
            Debug.Log(Nome + " é hostil!");
        }
        else
        {
            Falar(0);
        }
    }

    public void TornarHostil()
    {
        hostil = true;
    }
}